genie-agent/src/main/java/com/netflix/genie/agent/execution/locks/Lock.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.agent.execution.locks;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+/**
+ * A simple interface representing a closeable lock.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+public interface Lock extends Closeable {
```

Comment:
That was my first thought, but I saw that it does not implement Closeable. I was trying to stick to the paradigm of having less boiler plater code. So, wanted to use -  https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html - to avoid closing/unlocking in finally. Hence created and interface which extended Closeable

Reply:
To make that explicit I can call the interface `CloseableLock`. Thoughts?